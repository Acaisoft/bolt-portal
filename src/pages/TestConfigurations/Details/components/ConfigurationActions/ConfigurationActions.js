import React from 'react'
import PropTypes from 'prop-types'
import { withStyles, Button, Typography, Grid, Tooltip } from '@material-ui/core'
import { PlayArrow, Edit, Delete } from '@material-ui/icons'

import styles from './ConfigurationActions.styles'

export const ConfigurationActions = ({
  classes,
  onRun,
  onEdit,
  onDelete,
  isPerformed,
  isRunning,
}) => (
  <Grid container justify="center" alignItems="center">
    <Button
      classes={{
        root: classes.actionButton,
        label: classes.actionButtonLabel,
      }}
      variant="contained"
      disabled={isRunning}
      onClick={onRun}
    >
      <PlayArrow />
      <Typography variant="body2">Play</Typography>
    </Button>
    <Button
      classes={{
        root: classes.actionButton,
        label: classes.actionButtonLabel,
      }}
      variant="contained"
      onClick={onEdit}
    >
      <Edit />
      <Typography variant="body2">Edit</Typography>
    </Button>
    <Tooltip
      title={isPerformed ? "You can't delete a performed scenario." : ''}
      PopperProps={{
        classes: {
          popper: {
            opacity: 1,
          },
        },
      }}
    >
      <span>
        <Button
          aria-label="Delete scenario"
          classes={{
            root: classes.actionButton,
            label: classes.actionButtonLabel,
          }}
          variant="contained"
          disabled={isPerformed}
          onClick={onDelete}
        >
          <Delete />
          <Typography variant="body2">Delete</Typography>
        </Button>
      </span>
    </Tooltip>
  </Grid>
)

ConfigurationActions.propTypes = {
  classes: PropTypes.object,
  onChange: PropTypes.func,
  onDelete: PropTypes.func,
  onEdit: PropTypes.func,
  onRun: PropTypes.func,
}

ConfigurationActions.defaultProps = {
  onChange: () => {},
  onDelete: () => {},
  onEdit: () => {},
  onRun: () => {},
}

export default withStyles(styles)(ConfigurationActions)
