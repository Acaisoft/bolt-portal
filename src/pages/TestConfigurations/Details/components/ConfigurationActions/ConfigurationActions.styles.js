export default ({ breakpoints, spacing }) => ({
  actionButton: {
    margin: spacing.unit * 1,
    width: 120,
    height: 120,
    [breakpoints.down('sm')]: {
      width: 80,
      height: 80,
    },
    [breakpoints.down('md')]: {
      width: 100,
      height: 100,
    },
  },
  actionButtonLabel: {
    display: 'flex',
    flexDirection: 'column',
  },
})
