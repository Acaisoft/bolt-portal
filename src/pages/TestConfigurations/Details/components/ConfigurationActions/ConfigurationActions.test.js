import React from 'react'
import { shallow } from 'enzyme'
import { mocks } from '~utils/tests'

import { ConfigurationActions } from './ConfigurationActions'

const initConfigurationActions = overrides => {
  const mockProps = {
    classes: mocks.ClassesProxy,
  }
  const mockMethods = {
    onEdit: jest.fn(),
    onChange: jest.fn(),
    onPlay: jest.fn(),
  }
  const wrapper = shallow(
    <ConfigurationActions {...mockProps} {...mockMethods} {...overrides} />
  )
  return { mockProps, wrapper }
}

describe('global: ConfigurationActions', () => {
  it('renders without crashing', () => {
    const { wrapper } = initConfigurationActions()
    expect(wrapper).toBeTruthy()
  })

  it('should render as expected', () => {
    const { wrapper } = initConfigurationActions({})
    expect(wrapper).toMatchSnapshot()
  })
})
