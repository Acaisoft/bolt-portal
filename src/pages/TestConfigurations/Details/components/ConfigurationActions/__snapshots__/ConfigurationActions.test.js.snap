// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: ConfigurationActions should render as expected 1`] = `
<GridMock
  alignItems="center"
  container={true}
  justify="center"
>
  <ButtonMock
    classes={
      Object {
        "label": "actionButtonLabel",
        "root": "actionButton",
      }
    }
    onClick={[Function]}
    variant="contained"
  >
    <PlayArrowMock />
    <TypographyMock
      variant="body2"
    >
      Play
    </TypographyMock>
  </ButtonMock>
  <ButtonMock
    classes={
      Object {
        "label": "actionButtonLabel",
        "root": "actionButton",
      }
    }
    onClick={[MockFunction]}
    variant="contained"
  >
    <CalendarTodayMock />
    <TypographyMock
      variant="body2"
    >
      Change
    </TypographyMock>
  </ButtonMock>
  <ButtonMock
    classes={
      Object {
        "label": "actionButtonLabel",
        "root": "actionButton",
      }
    }
    onClick={[MockFunction]}
    variant="contained"
  >
    <EditMock />
    <TypographyMock
      variant="body2"
    >
      Edit
    </TypographyMock>
  </ButtonMock>
  <TooltipMock
    PopperProps={
      Object {
        "classes": Object {
          "popper": Object {
            "opacity": 1,
          },
        },
      }
    }
    title=""
  >
    <span>
      <ButtonMock
        aria-label="Delete scenario"
        classes={
          Object {
            "label": "actionButtonLabel",
            "root": "actionButton",
          }
        }
        onClick={[Function]}
        variant="contained"
      >
        <DeleteMock />
        <TypographyMock
          variant="body2"
        >
          Delete
        </TypographyMock>
      </ButtonMock>
    </span>
  </TooltipMock>
</GridMock>
`;
